### ISSUE TITLE
Got "Minion did not return" when running rabbitmq.list_users from master

### ISSUE THREAD

**Issue by enver** [saltstack/salt#31765]
### Setup

```
# salt 'standalone2' rabbitmq.list_users
standalone2:
    Minion did not return. [No response]
```
### Steps to Reproduce Issue
- Use rabbitmq formula to setup rabbitmq server.
- Run `salt 'standalone2' rabbitmq.list_users`

```
==> master <==
2016-03-09 13:38:27,583 [salt.master                                          ][INFO    ][29359] User root Published command rabbitmq.list_users with jid 20160309133827581746
2016-03-09 13:38:32,595 [salt.master                                          ][INFO    ][29361] User root Published command saltutil.find_job with jid 20160309133832591789
2016-03-09 13:38:32,734 [salt.utils.job                                       ][INFO    ][29383] Got return from standalone2 for job 20160309133832591789
2016-03-09 13:38:32,843 [salt.master                                          ][INFO    ][29360] User root Published command cmd.run with jid 20160309133832842318
2016-03-09 13:38:33,141 [salt.utils.job                                       ][INFO    ][29362] Got return from deploy for job 20160309133832842318
```

```
==> minion (debug) <==
2016-03-09 13:38:27,512 [salt.minion                                          ][INFO    ][13409] User root Executing command rabbitmq.list_users with jid 20160309133827581746
2016-03-09 13:38:27,515 [salt.minion                                          ][DEBUG   ][13409] Command details {'tgt_type': 'glob', 'jid': '20160309133827581746', 'tgt': 'standalone2', 'ret': '', 'user': 'root', 'arg': [], 'fun': 'rabbitmq.list_users'}
2016-03-09 13:38:27,540 [salt.minion                                          ][INFO    ][15633] Starting a new job with PID 15633
2016-03-09 13:38:27,547 [salt.utils.lazy                                      ][DEBUG   ][15633] LazyLoaded rabbitmq.list_users
2016-03-09 13:38:27,550 [salt.utils.lazy                                      ][DEBUG   ][15633] LazyLoaded cmd.run
2016-03-09 13:38:27,618 [salt.loaded.int.module.cmdmod                        ][INFO    ][15633] Executing command 'rabbitmqctl list_users' as user 'root' in directory '/root'
2016-03-09 13:38:27,725 [salt.loaded.int.module.cmdmod                        ][DEBUG   ][15633] output: Listing users ...
guest   [administrator]
test    [administrator]
...done.
2016-03-09 13:38:27,725 [salt.minion                                          ][DEBUG   ][15633] Minion return retry timer set to 9 seconds (randomized)
2016-03-09 13:38:27,726 [salt.minion                                          ][INFO    ][15633] Returning information for job: 20160309133827581746
2016-03-09 13:38:27,727 [salt.transport.zeromq                                ][DEBUG   ][15633] Initializing new AsyncZeroMQReqChannel for ('/etc/salt/pki/minion', 'standalone2', 'tcp://xxx.xxx.xxx.xxx:4506', 'aes')
2016-03-09 13:38:27,727 [salt.crypt                                           ][DEBUG   ][15633] Initializing new SAuth for ('/etc/salt/pki/minion', 'standalone2', 'tcp://xxx.xxx.xxx.xxx:4506')
2016-03-09 13:38:32,526 [salt.minion                                          ][INFO    ][13409] User root Executing command saltutil.find_job with jid 20160309133832591789
2016-03-09 13:38:32,526 [salt.minion                                          ][DEBUG   ][13409] Command details {'tgt_type': 'glob', 'jid': '20160309133832591789', 'tgt': 'standalone2', 'ret': '', 'user': 'root', 'arg': ['20160309133827581746'], 'fun': 'saltutil.find_job'}
2016-03-09 13:38:32,538 [salt.minion                                          ][INFO    ][15672] Starting a new job with PID 15672
2016-03-09 13:38:32,551 [salt.utils.lazy                                      ][DEBUG   ][15672] LazyLoaded saltutil.find_job
2016-03-09 13:38:32,552 [salt.minion                                          ][DEBUG   ][15672] Minion return retry timer set to 10 seconds (randomized)
2016-03-09 13:38:32,553 [salt.minion                                          ][INFO    ][15672] Returning information for job: 20160309133832591789
2016-03-09 13:38:32,553 [salt.transport.zeromq                                ][DEBUG   ][15672] Initializing new AsyncZeroMQReqChannel for ('/etc/salt/pki/minion', 'standalone2', 'tcp://xxx.xxx.xxx.xxx:4506', 'aes')
2016-03-09 13:38:32,554 [salt.crypt                                           ][DEBUG   ][15672] Initializing new SAuth for ('/etc/salt/pki/minion', 'standalone2', 'tcp://xxx.xxx.xxx.xxx:4506')
```
### Versions Report

Master and minion run same salt version

```
# salt --versions-report
Salt Version:
           Salt: 2015.8.7

Dependency Versions:
         Jinja2: 2.8
       M2Crypto: 0.20.2
           Mako: Not Installed
         PyYAML: 3.11
          PyZMQ: 14.5.0
         Python: 2.6.6 (r266:84292, Jul 23 2015, 15:22:56)
           RAET: Not Installed
        Tornado: 4.2.1
            ZMQ: 4.0.5
           cffi: Not Installed
       cherrypy: Not Installed
       dateutil: Not Installed
          gitdb: Not Installed
      gitpython: Not Installed
          ioflo: Not Installed
        libgit2: 0.20.0
        libnacl: 1.4.3
   msgpack-pure: Not Installed
 msgpack-python: 0.4.6
   mysql-python: Not Installed
      pycparser: Not Installed
       pycrypto: 2.6.1
         pygit2: 0.20.3
   python-gnupg: Not Installed
          smmap: Not Installed
        timelib: Not Installed

System Versions:
           dist: centos 6.7 Final
        machine: x86_64
        release: 2.6.32-573.el6.x86_64
         system: CentOS 6.7 Final
```

RMQ server version: 3.1.5
### Note

Running command on minion gives following results:

```
# salt-call rabbitmq.list_users
local:
    ----------
    guest:
    test:
[DEBUG   ] Error deleting fd from IOLoop
Traceback (most recent call last):
  File "/usr/lib64/python2.6/site-packages/tornado/ioloop.py", line 714, in remove_handler
    self._impl.unregister(fd)
  File "/usr/lib64/python2.6/site-packages/zmq/eventloop/ioloop.py", line 115, in unregister
    return self._poller.unregister(fd)
  File "/usr/lib64/python2.6/site-packages/zmq/sugar/poll.py", line 71, in unregister
    idx = self._map.pop(socket)
KeyError: 'pop(): dictionary is empty'
```

**Comment by enver** [comment]
Same issue mentioned in comments in #25125

**Comment by jfindlay** [comment]
@enver, thanks for reporting.  Is it possible for you to test the fix from @s-schweer?

@s-schweer, is it possible for you to backport your fix to 2015.8?  Thanks.

**Comment by enver** [comment]
@jfindlay, @s-schweer 
Sure, just let me know when it is available
